File: LeetCode/1/1-Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
      map<int, int> table;
      vector<int> result;
      vector<int>::iterator it = nums.begin();
      for(; it != nums.end(); ++it) {
        int index = it-nums.begin();
        int num = *it;
        int remain = target - num;
        if(table.find(num) != table.end()) {
          result.push_back(table[num]);
          result.push_back(index);
          break;
        } else {
          table[remain] = index;
        }
      }
      return result;
    }
};

File: LeetCode/1/1-Two_Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  var t = {};
  for(var i = 0; i < nums.length; i++) {
    var r = target-nums[i];
    if(typeof t[r] != 'undefined') return [t[r], i];
    else t[nums[i]] = i;
  }
};
